### DISCUSSION TITLE
x-mask allows trailing character (even an invalid one)

### DISCUSSION THREAD

**Discussion by jeffchown** [alpinejs/alpine#3818] [7. Bugs (archived)]
Hello,

I've been implementing `x-mask` in one of my applications (using Livewire v3.0.10 and Alpine v3.13.2) and have recently noticed that my inputs using x-mask allow the user to type a trailing character that is not shown in the input but is stored in the property linked via `wire:model`.

For example, using markup:

```
      <input
          wire:model="form.phone"
          x-mask="(999)999-9999"
          placeholder="(999)999-9999"
      />

```

if the user types `1a2b34567890` in the input, then x-mask properly constrains & formats the input to `(123)456-7890`.  If the user then continues to type and enters **any** character (number, letter, etc.), the character is not echoed in the input, but **is** stored in the `form.phone` property (and therefore saved to the db upon update).

I have tried many different masks in x-mask and the result is the same for all, they all allow a trailing character.

**Comment by SimoTod** [comment]
I think it's more a livewire bug because both mask and wire:model react to the input event.
You could (and maybe should if you don't have a good reason) use the lazy modifier so you don't save to the db on every keystroke.

**Reply by jeffchown** [reply]
Thanks for the idea @SimoTod - I have tried the `wire:model` with each of `.live`, `.lazy`, `.defer` and each lead to the same results. (My example above is written in Livewire v3 which defaults to `.defer`)

**Comment by ekwoka** [comment]
you can't have multiple of the same attribute on an element.
 
 x-mask definitely does not allow additional characters when used just with alpine.

**Reply by jeffchown** [reply]
Thanks for pointing out my typo @ekwoka - I only made the mistake when posting the issue, in my code, there is only one `x-mask`.
I have edited my initial post.

x-mask is not allowing any *visible* trailing characters when used with `wire:model`, but does store the trailing character (if one is typed) in the property.

It of course could be an issue when paired with Livewire, open to any ideas/findings.

**Reply by ekwoka** [reply]
My best guess is it s about with wire model that is still consuming the input...

**Reply by jeffchown** [reply]
@ekwoka Thanks re: your comment that `x-mask` does not allow a trailing character when used only with Alpine - I confirmed that using the following snippet:

```
        <div x-data="{ message: '' }">
            <input type="text" x-model="message" x-mask="(999)999-9999">

            <span x-text="message">
        </div>
```

When testing, any trailing characters were **not** added/shown in the `message` property.
So... it is definitely an issue when paired with Livewire.  I will move this discussion to the Livewire repo
